### ISSUE TITLE
The "Autoplay Media" setting is wrongly reset to the default value after update

### ISSUE THREAD

**Issue by Softvision-GeluHaiduc** [MozillaReality/FirefoxReality#1781]
**[Affected Versions]:**
- Firefox Reality 1.4-32481747 (RC3)

**[Affected Platforms]:**
- Oculus Go - Android 7.1.2
- Oculus Quest - Android 7.1.1
- HTC Vive Focus Plus - Android 7.1.2

**[Prerequisites]:**
- Have the production version of Firefox Reality 1.3 installed.
- Have the latest 1.4-32481747 (RC3) build downloaded.
- The sideloading is set up following the steps from: [link](https://github.com/MozillaReality/FirefoxReality/wiki/Developer-workflow:-install-ADB)
- The headset is turned on and the app is opened.

**[Steps to reproduce]:**
1. Open the FxR 1.3 app.
2. Click the "Settings" (gear) button from the menu tray.
3. Click the "Privacy & Security" card and turn the "Autoplay Media" setting on.
4. Press the oculus/vive button on the controller and "Quit" the app.
5. Sideload the following command to update the app: "adb install -r <path to the 1.4-32481747 (RC3) build from prerequisites>".
6. Open the FxR app again and click the "Settings" (gear) button from the menu tray.
7. Click the "Display" card and observe the "Autoplay Media" setting.

**[Expected results]:**
- The "Autoplay Media" setting is still turned on after the update.

**[Actual results]:**
- The "Autoplay Media" setting is reset to its default value, off.

**[Notes]:**
- Attached a screen recording of the issue [here](https://drive.google.com/file/d/1x68gafgeZWIoKWdIh3grDyCkDAoDomY1/view?usp=sharing).

**Comment by bluemarvin** [comment]
It appears that the autoplay preference is not save at all between runs and just killing FxR and restarting will loose the preference setting. Unfortunately the autoplay API is currently broken in GeckoView and needs to be fixed there before we can meaningfully address this. I'm going to move this to 1.5 as we will be blocked on GeckoView even if we are saving the autoplay state between runs.

**Comment by bluemarvin** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=1577596

**Comment by bluemarvin** [comment]
Will try and make it part of the 1.4.1 release already planned to pull in the top crasher fix in GeckoView:
https://bugzilla.mozilla.org/show_bug.cgi?id=1579028

**Comment by Softvision-GeluHaiduc** [comment]
I have verified this issue with an update from the last release 1.3 available in stores to the latest Firefox Reality 1.4-32500011 (RC4) installed on the Oculus Go, Quest and HTC Vive Focus Plus headsets. 
The issue is still reproducible, the "Autoplay Media" setting is wrongly reset to the default value after an update.
I've attached a new screen recording of the issue [here](https://drive.google.com/file/d/1Xkv9yW0zJKErZzt3j-CknV942py9fb_U/view?usp=sharing), and the logcat file [here](https://drive.google.com/file/d/1UIfOJklKOXE33avq1v7Oy5tqDRFoXL0P/view?usp=sharing).

**Comment by bluemarvin** [comment]
To be clear, 1.3 does not save the state. This is a bug in 1.3. So installing a new version is not relevant to the issue. Only rc4 and later will save the state. If going forward we lose state, that will be an issue.

**Comment by Softvision-GeluHaiduc** [comment]
I have verified that this issue is no longer reproducible with an update from 1.4 RC4 to to the latest Firefox Reality 1.4-32522317 (RC5) installed on the Oculus Go, Quest and HTC Vive Focus Plus headsets.
Now, the "Autoplay Media" setting is saved between 1.4 updates.
We will add the "[QA]:Wontfix issue" label to this as the 1.3 users will still be affected by this issue when updating to 1.4.
